### DISCUSSION TITLE
Highlights for ESMValCore v2.8.0 release

### DISCUSSION THREAD

**Discussion by remi-kazeroni** [ESMValGroup/ESMValCore#1954] [General]
@ESMValGroup/esmvaltool-developmentteam This discussion collects possible highlights for release v2.8.0 ⭐

Please add pull requests that you want to nominate as answers below, afterwards people can vote for their favorite new feature.

The top n (TBA) pull requests will be highlighted in the upcoming release!

Rules:

- _Exactly_ one (group of) PR per answer
- Only already merged PRs are allowed.

  - List of allowed PRs: https://github.com/ESMValGroup/ESMValCore/pulls?page=4&q=is%3Apr+merged%3A%3E2022-10-14+-label%3Acondalock+base%3Amain+sort%3Acreated-asc 
  - First draft of the Changelog (WIP): https://esmvaltool--1952.org.readthedocs.build/projects/ESMValCore/en/1952/changelog.html

**Comment by remi-kazeroni** [comment]
@bouweandela 's PR #1609 "Godzilla"

_Originally posted by @valeriupredoi in https://github.com/ESMValGroup/ESMValCore/issues/1952#issuecomment-1452204650_

**Comment by valeriupredoi** [comment]
cheers for opening this @remi-kazeroni and my apologies for jumping the gun by posting about this on the Issue instead of a Discussion - I'll let you be the capt'n of this Release from now on :anchor: And for the others, in case you need a visual for #1609 - here is one 
![51eQiq](https://user-images.githubusercontent.com/28983971/222716124-557bf9df-372f-43f4-9099-660cc9a6cb7b.jpg)

**Comment by bouweandela** [comment]
Extension of the Python API would be my proposed highlight, with the addition of the new modules

- [esmvalcore.dataset](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/api/esmvalcore.dataset.html)
- [esmvalcore.config](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/api/esmvalcore.local.html)
- [esmvalcore.local](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/api/esmvalcore.config.html)

and example notebooks available here: https://docs.esmvaltool.org/projects/esmvalcore/en/latest/example-notebooks.html

most of it implemented in #1877, #1835, #1816. If I really have to pick only one addition, it would be the addition of the `esmvalcore.dataset.Dataset` class because this was the missing link between the input files and using the preprocessor functions.

**Reply by remi-kazeroni** [reply]
I'm fine listing more than PR per highlight in cases like this. It actually looks more appealing to advertise that we have improved the API with the following modules rather than listing just one module.

**Comment by schlunma** [comment]
I nominate https://github.com/ESMValGroup/ESMValCore/pull/1765, which implemented proper support for CORDEX datasets :tada:

**Comment by remi-kazeroni** [comment]
Extensions of the `multi_model_statistics` preprocessor. This was implemented in a series of PRs:

- https://github.com/ESMValGroup/ESMValCore/pull/1808
- https://github.com/ESMValGroup/ESMValCore/pull/1813
- https://github.com/ESMValGroup/ESMValCore/pull/1849
- https://github.com/ESMValGroup/ESMValCore/pull/1921
- https://github.com/ESMValGroup/ESMValCore/pull/1934

**Comment by remi-kazeroni** [comment]
I also nominate #1664 which made ICON output UGRID-compatible on-the-fly

**Comment by remi-kazeroni** [comment]
I propose to leave this open until **Tuesday, March 14 noon**. This is the deadline to propose PR but all to upvote your favourite PRs or groups of PRs. This would help me to sort them for the Highlights section of the changelog. Thanks 👍

**Comment by bouweandela** [comment]
Excited to see that we have so many improvements for the upcoming release!

**Reply by remi-kazeroni** [reply]
Me too! I'm planning to (try to) add these 5 highlights to the changelog. I think it would reflect the diversity of contributions we have received for this release and gives a good overview of the novelties in v2.8. Feel free to continue upvoting or add emojis to your favourite ones. It would help me to rank the ones below #1609 😅

**Comment by valeriupredoi** [comment]
yeah it's like that song by The Weather Girls - "It's Raining PRs" :laughing:

**Comment by remi-kazeroni** [comment]
Thanks for all the suggestions and votes. The highlights are made available in #1959, see [documentation](https://esmvaltool--1959.org.readthedocs.build/projects/ESMValCore/en/1959/changelog.html).
